### PR TITLE
Add CODEOWNERS

### PR DESCRIPTION
**Pull request by cdrage**
This adds GitHub code owners to automatically ping the appropriate
maintainers when new PR's are opened.

See: https://help.github.com/articles/about-code-owners/

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+/pkg/auth/ @anmolbabu @girishramnani 
+/pkg/catalog/ @surajnarwade @girishramnani @metacosm @geoand 
+/pkg/component/ @anmolbabu  @mohammedzee1000 
+/pkg/config/ @syamgk  @mohammedzee1000 
+/pkg/log/ @girishramnani  @cdrage
+/pkg/notify/ @girishramnani  @cdrage
+/pkg/occlient/ @kadel @metacosm @geoand  @cdrage
+/pkg/odo/ @anmolbabu  @geoand 
+/pkg/project/ @mik-dass  @mohammedzee1000 
+/pkg/secret/ @girishramnani
```
Comment (kadel) [inline CODEOWNERS]: missing geoand

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/pkg/catalog/ @surajnarwade @girishramnani @metacosm @geoand 
+/pkg/component/ @anmolbabu  @mohammedzee1000 
+/pkg/config/ @syamgk  @mohammedzee1000 
+/pkg/log/ @girishramnani  @cdrage
+/pkg/notify/ @girishramnani  @cdrage
+/pkg/occlient/ @kadel @metacosm @geoand  @cdrage
+/pkg/odo/ @anmolbabu  @geoand 
+/pkg/project/ @mik-dass  @mohammedzee1000 
+/pkg/secret/ @girishramnani 
+/pkg/service/ @surajnarwade
```
Comment (kadel) [inline CODEOWNERS]: missing geoand

**Comment by cdrage** [comment]
Added the new change based on the comments :+1:

File: CODEOWNERS [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# Odo specific
+/pkg/application/ @syamgk  @cdrage
+/pkg/auth/ @anmolbabu @girishramnani 
+/pkg/catalog/ @surajnarwade @girishramnani @metacosm @geoand 
+/pkg/component/ @anmolbabu  @mohammedzee1000 
+/pkg/config/ @syamgk  @mohammedzee1000 
+/pkg/log/ @girishramnani  @cdrage
+/pkg/notify/ @girishramnani  @cdrage
+/pkg/occlient/ @kadel @metacosm @geoand  @cdrage
+/pkg/odo/ @anmolbabu  @geoand
```
[the remark this answers is not in the document]
Comment (metacosm) [reply CODEOWNERS]: Can you add me to this package as well, please?

**Comment by metacosm** [comment]
Let's merge once CI is green!

**Pull request merged**